File: chat-uikit/src/main/java/io/agora/chat/uikit/chat/interfaces/OnMessageItemClickListener.java
package io.agora.chat.uikit.chat.interfaces;

import android.view.View;

import io.agora.chat.ChatMessage;

public interface OnMessageItemClickListener {
    /**
     * Click on the message bubble area
     * @param message
     * @return
     */
    boolean onBubbleClick(ChatMessage message);

    /**
     * Long press the message bubble area
     * @param v
     * @param message
     * @return
     */
    boolean onBubbleLongClick(View v, ChatMessage message);

    /**
     * Click on the avatar
     * @param username
     */
    void onUserAvatarClick(String username);

    /**
     * Long press on the avatar
     * @param username
     */
    void onUserAvatarLongClick(String username);

    /**
     * Click on thread region
     * @param messageId
     * @param threadId
     * @return
     */
    default boolean onThreadClick(String messageId, String threadId) {
        return false;
    }

    /**
     * Long press on thread region
     * @param v
     * @param messageId
     * @param threadId
     * @return
     */
    default boolean onThreadLongClick(View v, String messageId, String threadId) {
        return false;
    }
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/feature/chat/interfaces/OnMessageItemClickListener.kt
package io.agora.uikit.feature.chat.interfaces

import android.view.View
import io.agora.uikit.common.ChatMessage

interface OnMessageItemClickListener {
    /**
     * Click on the message bubble area
     * @param message
     * @return
     */
    fun onBubbleClick(message: ChatMessage?): Boolean

    /**
     * Long press the message bubble area
     * @param v
     * @param message
     * @return
     */
    fun onBubbleLongClick(v: View?, message: ChatMessage?): Boolean

    /**
     * Click the resend view.
     * @param message
     * @return
     */
    fun onResendClick(message: ChatMessage?): Boolean

    /**
     * Click on the avatar
     * @param userId
     */
    fun onUserAvatarClick(userId: String?)

    /**
     * Long press on the avatar
     * @param userId
     */
    fun onUserAvatarLongClick(userId: String?)

}